src/network/networkd-dhcp-common.c

```diff
@@ -227,6 +227,73 @@ int config_parse_iaid(const char *unit,
         return 0;
 }
 
+int config_parse_dhcp6_pd_length(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        uint8_t u;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = safe_atou8(rvalue, &u);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Failed to parse IPv6PrefixDelegationLength=%s, ignoring assignment: %m", rvalue);
+                return 0;
+        }
+
+        network->dhcp6_pd_length = u;
+
+        return 0;
+}
+
+int config_parse_dhcp6_pd_address(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = data;
+        union in_addr_union address;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = in_addr_from_string(AF_INET6, rvalue, &address);
```

Comment:
`r = in_addr_from_string(AF_INET6, rvalue, (in_addr_union *) &network->dhcp6_pd_address);`
Then, unnecessary `memcpy()` can be dropped.

Reply:
dropped